realm-annotations-processor/src/test/resources/io/realm/RealmDefaultModuleMediator.java

```diff
@@ -0,0 +1,134 @@
+package io.realm;
+
+
+import android.util.JsonReader;
+import io.realm.exceptions.RealmException;
+import io.realm.internal.ImplicitTransaction;
+import io.realm.internal.RealmObjectProxy;
+import io.realm.internal.RealmProxyMediator;
+import io.realm.internal.Table;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONException;
+import org.json.JSONObject;
+import some.test.AllTypes;
+
+@io.realm.annotations.internal.RealmModule
+class DefaultRealmModuleMediator
+        implements RealmProxyMediator {
+
+    private static final List<Class<? extends RealmObject>> MODEL_CLASSES;
+    static {
+        List<Class<? extends RealmObject>> modelClasses = new ArrayList<Class<? extends RealmObject>>();
+        modelClasses.add(AllTypes.class);
+        MODEL_CLASSES = Collections.unmodifiableList(modelClasses);
+    }
+
+    @Override
+    public Table createTable(Class<? extends RealmObject> clazz, ImplicitTransaction transaction) {
+        if (clazz == null) throw new NullPointerException("A class extending RealmObject must be provided");
+
+        if (clazz.equals(AllTypes.class)) {
+            return AllTypesRealmProxy.initTable(transaction);
+        } else {
+            throw new RealmException("Could not find the generated proxy class for " + clazz + ". " + RealmProxyMediator.APT_NOT_EXECUTED_MESSAGE);
```

Comment:
Same line many places - make private method. Same for initial check for clazz == null